k8s/solr/README.md

```diff
@@ -0,0 +1,417 @@
+
+> Visit [Monitoring Solr with Prometheus and Grafana documentation](https://lucene.apache.org/solr/guide/8_6/monitoring-solr-with-prometheus-and-grafana.html) for more information.
+
+To check metrics from Solr exporter service forward exporter service port to local machine
+```shell
+# Forward Solr exporter service port to local workspace
+kubectl port-forward svc/${APP_INSTANCE_NAME}-solr-exporter --namespace ${NAMESPACE} 9983
+```
+Visit http://localhost:9983/metrics to see metrics from Solr.
+
+# Scaling
+
+## Scaling the cluster up or down
+
+By default, the SolrCloud distributed cluster starts with 3 replicas.
+
+To change the number of replicas, use the following command, where `REPLICAS`
+is your desired number of replicas:
+
+```
+export REPLICAS=4
+kubectl scale statefulsets "${APP_INSTANCE_NAME}-solr" \
+  --namespace "${NAMESPACE}" --replicas="${REPLICAS}"
+```
+
+When this option is used to scale down a cluster, it reduces the number of
+replicas without disconnecting nodes from the cluster. Scaling down also does
+not affect the `persistentvolumeclaims` of your StatefulSet.
+
+> Visit SolrCloud [Autoscaling documentation](https://lucene.apache.org/solr/guide/8_6/solrcloud-autoscaling-auto-add-replicas.html) to enable autoscaling for collections to replication collections automatically to new replicas.
+
+# Upgrading the app
+
+Before you upgrade the app, we recommend that you back up all of your
+SolrCloud collections.
+> Visit [Solr Upgrade Notes](https://lucene.apache.org/solr/guide/8_6/solr-upgrade-notes.html) to check limitations for upgrading between versions
+
+The Solr StatefulSet is configured to roll out updates automatically. To
+start the update, patch the StatefulSet with a new image reference, where
+`[NEW_IMAGE_REFERENCE]` is the Docker image reference of the new image that you
+want to use:
+
+```shell
+kubectl set image statefulset ${APP_INSTANCE_NAME}-solr --namespace ${NAMESPACE} \
+  "solr=[NEW_IMAGE_REFERENCE]"
+```
+
+To check the status of Pods in the StatefulSet, and the progress of the
+new image, run the following command:
+
+```shell
+kubectl get pods --selector app.kubernetes.io/name=${APP_INSTANCE_NAME} \
+  --namespace ${NAMESPACE}
+```
+# Solr Standalone mode
+It is also possible to deploy this Solr application as a standalone mode for testing purposes.
+> Warning: This should not be used for production!
+
+With Standalone mode enabled, instead of external ZooKeeper ensemble embedded ZooKeeper will be used and Solr will be deployed with single replica only.
+Also no any authentication will be applied to the solution to make it easier for testing.
+
+To deploy Solr as a standalone mode, run following commands:
+```shell
+# Export necessary variables
```

Comment:
I see here you add all variables already defined before, it's fine as we're talking about another solution, but I would also add `TAG` var.